### PR TITLE
Support empty blendshape targets

### PR DESCRIPTION
**Pull request by ysiewappl**
Hi:

This PR supplements the previous added support for exporting blendshapes with some minor changes to the names of the targets exported, and also adds support for exporting empty blendshape targets, gated by the `ignoreWarnings` flag.

Please review and raise any concerns.

### CODE REVIEW

File: lib/usd/translators/meshWriter_BlendShapes.cpp [hunk 1]
```diff
@@ -636,8 +674,13 @@ MObject PxrUsdTranslators_MeshWriter::writeBlendShapeData(UsdGeomMesh& primSchem
                         CHECK_MSTATUS_AND_RETURN(stat, MObject::kNullObj);
                         // NOTE: (yliangsiew) Because a single weight can drive multiple targets, we
                         // have to put a numeric suffix in the target name.
```
Comment (mattyjams) [inline lib/usd/translators/meshWriter_BlendShapes.cpp:676]: Is this comment still accurate now? Or maybe it's only accurate in a certain condition (`k != 0`)?

File: lib/usd/translators/meshWriter_BlendShapes.cpp [hunk 2]
```diff
@@ -804,12 +849,20 @@ MObject PxrUsdTranslators_MeshWriter::writeBlendShapeData(UsdGeomMesh& primSchem
...
```
Comment (mattyjams) [inline lib/usd/translators/meshWriter_BlendShapes.cpp:851]: Similar here. These could maybe just move into the `else` blocks?

**Review by mattyjams — APPROVED** [review]
I'm still not particularly well-versed in blend shapes, but seems reasonable to me!

File: lib/usd/translators/meshWriter_BlendShapes.cpp [hunk 3]
```diff
@@ -505,19 +527,31 @@ void findUnionAndProcessArrays(
...
         const VtVec3fArray& origOffsetsArray = offsetsArrays[i];
         const size_t        numOrigOffsets = origOffsetsArray.size();
-        TF_VERIFY(numOrigOffsets != 0);
-        VtVec3fArray& newOffsetsArray = unionOffsetsArrays[i];
+        VtVec3fArray&       newOffsetsArray = unionOffsetsArrays[i];
         newOffsetsArray.assign(numUnionIndices, GfVec3f(0.0f));
 
         const VtVec3fArray& origNormalsArray = normalsArrays[i];
+        const size_t        numOrigNormals = origNormalsArray.size();
 #ifdef _DEBUG
```
Comment (williamkrick) [inline lib/usd/translators/meshWriter_BlendShapes.cpp:535]: I think we don't need the _DEBUG preprocessor macro anymore.

**Review by williamkrick — APPROVED** [review]
It makes sense to me. Do you have a lot of scenes with no-op blend shape targets? I don't understand the circumstances which would generate Maya scenes that cause this code path to be used. Clearly it is happening because you've found you need it, I just don't have the imagination to figure out why.

Can you share the workflow that produces this type of blend shape?

**Comment by ysiewappl** [comment]
> It makes sense to me. Do you have a lot of scenes with no-op blend shape targets? I don't understand the circumstances which would generate Maya scenes that cause this code path to be used. Clearly it is happening because you've found you need it, I just don't have the imagination to figure out why.
> 
> Can you share the workflow that produces this type of blend shape?

We have a need by our internal artists that requires "placeholder" blendshapes be present, so that client code in other applications used for triggering various shapes across different USD files is able to at least find an empty delta to trigger (even if the shape isn't explicitly authored). It's arguable that this should be on client-side parsing code to handle...but Maya itself also allows for "empty" targets to be created via the GUI itself by the average non-technical end-user (i.e. duplicate a cube twice and apply the second cube as a blendshape target of the first without making any topology changes), so we thought it was best to kill two birds with one stone in this case and support both use cases.

File: lib/usd/translators/meshWriter_BlendShapes.cpp (hunk 3 above)
Comment (ysiewappl) [reply lib/usd/translators/meshWriter_BlendShapes.cpp]: Resolved in 522b6b0.

File: lib/usd/translators/meshWriter_BlendShapes.cpp (hunk 2 above)
Comment (ysiewappl) [reply lib/usd/translators/meshWriter_BlendShapes.cpp]: Resolved in 522b6b0.

File: lib/usd/translators/meshWriter_BlendShapes.cpp (hunk 1 above)
Comment (ysiewappl) [reply lib/usd/translators/meshWriter_BlendShapes.cpp]: Resolved in 522b6b0.

**Comment by kxl-adsk** [comment]
@ysiewappl I was about to merge this change, but how about we add a test to cover the improvement you made?

**Comment by ysiewappl** [comment]
> @ysiewappl I was about to merge this change, but how about we add a test to cover the improvement you made?

This is addressed in 824510a.

**Pull request merged**